app/sqs.go

```diff
@@ -32,6 +37,43 @@ type Message struct {
 	Retry                  int
 	MessageAttributes      map[string]MessageAttributeValue
 	GroupID                string
+	SentTime			   time.Time
+}
+
+func (m *Message) IsReadyForReceipt() bool {
+	randomLatency, err := getRandomLatency()
+	if err != nil {
+		log.Error(err)
+		return true
+	}
+	return m.SentTime.Add(randomLatency).Before(time.Now())
+}
+
+func getRandomLatency() (time.Duration, error){
+	minVar := os.Getenv("GOAWS_RANDOM_LATENCY_MIN")
```

Comment:
Sorry this took so long to evaluate but the values should be in the goaws.conf not in environment variables.  This is for consistency sake with how all configs are done.  thanks!

Reply:
@p4tin no worries, I've just pushed another commit - is this what you had in mind?